src/main/scala/beam/utils/FileUtils.scala

```diff
@@ -193,6 +193,35 @@ object FileUtils extends LazyLogging {
     new BufferedReader(new InputStreamReader(new UnicodeInputStream(stream), StandardCharsets.UTF_8))
   }
 
+  def readerFromIterator(iterator: Iterator[String]): java.io.Reader = {
```

Comment:
I think this is an old piece of code and I didn't add it. How would you suggest to replace it?

Reply:
Since this is an old piece of code which was just moved into a new file let's leave it. Ultimately what I was saying is that the only usage is that of an iterator -> reader -> iterator ... so get rid of the back and forth and just use the iterator. However the second iterator is more csv related, so it does change a little....it seems overly complex, but not worth touching in this change maybe